### ISSUE TITLE
[Question] Action space seeded outside make_vec_env(seed=seed)

### ISSUE THREAD

**Issue by leobxpan** [DLR-RM/stable-baselines3#1848]
### ❓ Question

When feeding my policy with the same observations, the policy always returns the same sampled action (`DiagGaussianDistribution.sample()`), despite different seeds provided when the environment is created with `make_vec_env(seed=seed)`, which should seed the action space differently. This happens when I'm using an initialized or a pre-trained model. And I've checked that when `OnPolicyAlgorithm._setup_model()` (https://github.com/DLR-RM/stable-baselines3/blob/a8e905977f3073066eb332f063f6335f355c455a/stable_baselines3/common/on_policy_algorithm.py#L113) is called, `self.seed` is None, so `self.set_random_seed()` immediately returns and does not seed everything. Any idea what caused this? Thank you in advance.

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014)
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by leobxpan** [comment]
In a word, I think the `np_random` is correctly handled through `make_vec_env()` but not pytorch seeding, which is set somewhere hidden. Because I use `np_random` to sample the dataset index, and seeding differently through `make_vec_env()` behaves as expected, but the sampled action is always the same given the same observation. However, I found that when I do not pass a seed to `make_vec_env()` (hence it defaults to none and seed should not be set), `np_random` still returns consistent results but with a different unknown seed.

**Comment by araffin** [comment]
Hello,
being the seen, `make_vec_env()` only prepare a call for `env.reset(seed=...)`, the rest (pytorch, action space, ...) is handled by `self.set_random_seed()`.
Is that your question/remark?
